src/invidious/comments/youtube.cr

```diff
@@ -131,89 +133,148 @@ module Invidious::Comments
                   node_replies = node["replies"]["commentRepliesRenderer"]
                 end
 
-                if node["comment"]?
-                  node_comment = node["comment"]["commentRenderer"]
-                else
-                  node_comment = node["commentRenderer"]
-                end
-
-                content_html = node_comment["contentText"]?.try { |t| parse_content(t, id) } || ""
-                author = node_comment["authorText"]?.try &.["simpleText"]? || ""
-
-                json.field "verified", (node_comment["authorCommentBadge"]? != nil)
+                if node["commentViewModel"]?
+                  cvm = node.dig("commentViewModel", "commentViewModel")
+                  comment_key = cvm["commentKey"]
+                  toolbar_key = cvm["toolbarStateKey"]
+                  if mutations.size != 0
```

Comment:
You should check if the mutations array is empty right after creating the array, rather than on each iteration. That way, you can return an empty response (and/or an error message) before doing any further processing.

I'm pretty sure that we'll never get the old format again.

Reply:
I'll remove the check for mutations.size as it should never be empty when we have a commentViewModel. I was able to get the old format a couple of days ago when I was testing this code + the old format is used for community posts.

Reply:
I got the old format for comments again today